ydb/core/tx/schemeshard/olap/operations/alter_table.cpp

```diff
@@ -265,6 +265,12 @@ class TAlterColumnTable: public TSubOperation {
 
         auto result = MakeHolder<TProposeResponse>(NKikimrScheme::StatusAccepted, ui64(OperationId.GetTxId()), ui64(ssId));
 
+        const bool isAlterSharding = Transaction.HasAlterColumnTable() && Transaction.GetAlterColumnTable().HasReshardColumnTable();
+        if (isAlterSharding && !AppData()->FeatureFlags.GetEnableAlterShardingInColumnShard()) {
```

Comment:
Its ok to check flag like this: here and at that point in time.

There is a single caveat though: `AppData()` is a global immutable object, and `EnableAlterShardingInColumnShard`s value would not be affected by changes made through dynamic config updates.